AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/ModelSyncedEventEmitter.swift

```diff
@@ -116,9 +116,10 @@ final class ModelSyncedEventEmitter {
         case .enqueued:
             recordsReceived += 1
         case .finished:
-            initialSyncOperationFinished = true
-            if recordsReceived == 0 {
+            if recordsReceived == 0 || recordsReceived == reconciledReceived {
                 sendModelSyncedEvent()
+            } else {
+                initialSyncOperationFinished = true
```

Comment:
to move this PR out of draft, let's add a unit test that mimics the behavior of the ordering of sync and reconcile events like so:

- sync record received (records received = 1)
- reconcile event applied (records reconciled = 1)
- sync record finished (causes the model synced event to be emitted)

We can also run this unit test against the previous code to see that it would fail to emit the model synced event.